src/matching/Generic_vs_generic.ml

```diff
@@ -1165,7 +1165,46 @@ and m_for_or_if_comp a b =
   | G.CompIf _, _ ->
       fail ()
 
+(* This just does some special coercions for Terraform.
+   If not a Terraform language, skip straight to `m_literal_inner`.
+   https://developer.hashicorp.com/terraform/language/expressions/type-constraints
+*)
 and m_literal a b =
+  Trace_matching.(if on then print_literal_pair a b);
+  fun tin ->
+    if Lang.equal tin.lang Lang.Hcl then
+      (* Is it ugly? Yes.
```

Comment:
Not very useful comment.

Reply:
Or did you mean that you could factorize with
| G.Int (x1, x2), B.String (y1, y2)
| G.String (y1, y2), G.Int (x1, x2) ->
  ???
Why not, since the body is the same? Or is it because
you want to distinguish when the String is from a pattern or from a target?
(possibly important since one can contain metavariable I guess).

Reply:
fair, i will leave a comment just saying we preserve order